packages/core/core/src/core-api/controller/transform.ts

```diff
@@ -51,7 +62,14 @@ function transformComponent(
     return data.map((datum) => transformComponent(datum, component));
   }
 
-  return transformEntry(data, component);
+  const res = transformEntry(data, component);
+
+  if (isNil(res)) {
+    return res;
+  }
+
+  const { id, documentId, attributes } = res;
+  return { id, documentId, ...attributes };
```

Comment:
just in case! is keeping components without the "attributes" nesting intended?

Reply:
Nope nice catch !

Reply:
I'll add a unit test for it

Reply:
oh you are right! wonder how it was "unnested" before